### DISCUSSION TITLE
Crazyflie not being scanned.

### DISCUSSION THREAD

**Discussion by akshaykumarsunkara** [bitcraze/discussions#1375] [Q&A]
Hello,

I've been trying to fix this for a while so I'd really appreciate any help.

Basically, I just set up my crazy radio and the crazyflie, and it all seems to work perfectly. Whenever I'm scanning for the crazyflie, the crazyradio flashes red and stops as soon as it stops scanning.

My crazyflie's M2 and M3 are constantly blue and the M1 is flashing red.

Whenever I cold boot it to install the firmware, I follow the procedure to bootload the Crazyflie and it works - the Crazyradio goes green and the crazyflie's propellers activate in sequence and then the Crazyradio stops being green and the Crazyflie goes back to its default state of flashing red in M1 and constantly staying blue in M2 and M3.

However, when I scan, I get no Crazyflie.

I heard that this might be because the addresses are different and I was told to go to connect/configure or something, however, I can't do this

![image](https://github.com/bitcraze/discussions/assets/30661280/492522be-186b-415e-87d5-66fc3ed27d41)

Can someone PLEASE help me!!!

**Comment by akshaykumarsunkara** [comment]
Okay, now when I'm flashing new firmware and flash the bolt release instead of the cf2 file, the drone is able to scan. However, it the M2 is constantly blue and it doesn't connect to the computer. Whenever I flash the cf2 file, the crazyradio is not able to scan the drone but the drones M2 and M3 are constantly lit blue and the M1 is flashing red. PLEASE can someone help.

**Reply by akshaykumarsunkara** [reply]
Also, when I flash it with flapper, the drone just starts its propellers and begins thrust.

**Reply by akshaykumarsunkara** [reply]
I've seen a lot of people saying to connect it with the USB-C but that doesn't work and still doesnt scan. I tried using the VMware as well - it doesn't work.

**Comment by ataffanel** [comment]
All the LED behavior you describe are good and are expected when the Crazyflie is working well.

If the Crazyflie is brand new it should not have a different address and after clicking "scan", it should appear in the list and you should be able to connect to it.

Otherwise, the solution to setup the address is indeed to connect Crazyflie over USB, make sure the USB driver is well installed with Zadig, and then click scan. The USB connection should appear in the list and you should be able to connect to it. Once connected, the configure 2.x menu will be enabled.

**Reply by akshaykumarsunkara** [reply]
Yes, I've had many people saying that I should try connecting it over USB and then configuring it. However, I'm not able to this either as the drone doesn't get detected.

This is the image of the device manager:

![image](https://github.com/bitcraze/discussions/assets/30661280/51913b2b-9250-4775-886a-fc8849792503)

I've installed the Crazyradio with Zadig's libusb0 (v1.2.7.3) and the Crazyflie with WinUSB (v6.1.7600.16385)

This is a picture of the console when it's scanning:

![image](https://github.com/bitcraze/discussions/assets/30661280/a4f65bab-a88e-425d-80dc-b84b5183f1d0)

**Reply by akshaykumarsunkara** [reply]
UPDATE: I installed a virtual machine and am running Cfclient from there. I was told to install libusb using Pip3, which I did, and I'm now able to scan usb://0. However, its just not able since my virtual machine keeps freezing. Any fixes?

**Reply by akshaykumarsunkara** [reply]
@ataffanel If I don't use my VMWare and go directly using my vscode, I'm not able to scan the drone at all, even via usb. Is there something wrong with the drivers I installed? For Crazyradio, I installed the libUsbK driver and it shows on the device manager. I deleted the driver that I installed for the crazyflie as I don't think I'm supposed to install any drivers for it. On device manager, it comes up under Other devices>Crazyflie 2.0. Is there any fix for this?

**Reply by akshaykumarsunkara** [reply]
UPDATE: I finally am able to scan and connect to USB://0. However, when I disconnect it, it doesn't work. I clicked on Configure 2.X and changed the address to match the one on cfclient. Here is a picture:



![image](https://github.com/bitcraze/discussions/assets/30661280/a46b9ae6-d513-418b-9a38-9465ef11d95b)


Please let me know how to do this because I think im like one step away from solving this. Thanks a lot.

**Reply by akshaysunkara** [reply]
UPDATE: I finally did it. it works perfectly!

**Reply by gemenerik** [reply]
Glad to hear you got it to work. Perhaps you can share what fixed your problem for people that stumble upon this discussion in the future.

**Comment by Robben-Ge** [comment]
I encountered the same issue where my Crazyflie wouldn't connect to the Crazyradio. However, I found the solution!

The problem was related to the radio bandwidth setting. Initially, it was set to 250 kbit/s, which caused the connection to fail. Once I changed the radio bandwidth to 1 Mbit/s, the Crazyflie was able to scan and connect successfully.

So, if you're facing a similar issue, I recommend adjusting the radio bandwidth from 250 kbit/s to 1 Mbit/s in the settings.

Hope this helps!
